src/pubsub/mod_pubsub_db.erl

```diff
@@ -124,7 +124,20 @@
 -callback remove_all_items(Nidx :: mod_pubsub:nodeIdx()) ->
     ok.
 
+-callback get_items(Nidx :: mod_pubsub:nodeIdx()) ->
+    {result, {[mod_pubsub:pubsubItem()], none}}.
+
+-callback get_item(Nidx :: mod_pubsub:nodeIdx(), ItemId :: mod_pubsub:itemId()) ->
+    {result, mod_pubsub:pubsubItem()} | {error, exml:element()}.
```

Comment:
I think DB backend shouldn't return XML in error tuple. Again - it should be node's or even main module's responsibility.

Reply:
I think so too, only the code depends on it so much.... We could refactor the whole pubsub infinitely...